cmd/buildah/main.go

```diff
@@ -118,6 +120,10 @@ func init() {
 	if err := rootCmd.PersistentFlags().MarkHidden("memory-profile"); err != nil {
 		logrus.Fatalf("unable to mark memory-profile flag as hidden: %v", err)
 	}
+	rootCmd.PersistentFlags().BoolVar(&globalFlagResults.registerBinfmt, "register-binfmt", os.Getenv("container") == "oci", "attempt to register binfmt_misc interpreters")
```

Comment:
If I run buildah within a podman container container=podman, should we handle this case?

Is "oci" set hen inside of Kubernetes?

Reply:
In my spot-check, podman sets it to "oci" when using either runc or crun as the runtime.  Do you have more information about how you're invoking podman, and which image you're using?

Reply:
I was checking with podman 5.2.2 on Fedora 40, running quay.io/buildah/stable as an unprivileged user.

Reply:
$ podman run alpine printenv container
podman
$ podman run fedora printenv container
oci
$ podman run ubuntu printenv container
podman

Depends on if the image defines the container environment variable I believe.  I think you could just check if the container environment variable is set.

Docker does not set it at all but does create the /.dockerenv file.  We create the /run/.containerenv file.

I think there is also a way to look at pid1 and figure it out which is what systemd does.

Reply:
Oh, ugh.  `skopeo inspect --config docker://registry.fedoraproject.org/fedora:40` does, indeed, include "container=oci" in its image config.